engine/src/main/java/io/zeebe/engine/processing/bpmn/gateway/ExclusiveGatewayProcessor.java

```diff
@@ -57,77 +47,80 @@ public ExclusiveGatewayProcessor(final BpmnBehaviors behaviors) {
   }
 
   @Override
-  public void onActivating(
-      final ExecutableExclusiveGateway element, final BpmnElementContext context) {
+  public void onActivate(
+      final ExecutableExclusiveGateway element, final BpmnElementContext activating) {
     if (element.getOutgoing().isEmpty()) {
       // there are no flows to take: the gateway is an implicit end for the flow scope
-      stateTransitionBehavior.transitionToActivated(context);
-      return;
+      final var activated = stateTransitionBehavior.transitionToActivated(activating);
+      final var completing = stateTransitionBehavior.transitionToCompleting(activated);
+      final var completed = stateTransitionBehavior.transitionToCompleted(completing);
```

Comment:
I thought about that, but at the time I didn't like it because it doesn't match well with the level of abstraction. It felt easier to understand what happens on processing of this command when all 3 steps are explicitly taken. But I can try it again